mongo/cursor.go

```diff
@@ -67,6 +68,47 @@ func newEmptyCursor() *Cursor {
 	return &Cursor{bc: driver.NewEmptyBatchCursor()}
 }
 
+// NewCursorFromDocuments creates a new Cursor pre-loaded with the provided documents, error and registry. If no registry is provided,
+// bson.DefaultRegistry will be used.
+//
+// The documents parameter must be a slice of documents. The slice may be nil or empty, but all elements must be non-nil.
+func NewCursorFromDocuments(documents []interface{}, err error, registry *bsoncodec.Registry) (*Cursor, error) {
```

Comment:
Using a `[]interface{}` to pass many documents is the same as the `InsertMany` API, but it can be difficult to use and frequently requires copying collections of documents into an `[]interface{}`. Would it be better to make the `documents` param an `interface{}` and use reflection to determine if the type is a slice or keep the API that's similar to `InsertMany`?

Reply:
Ah interesting point. Discussed offline: it does seem that accepting an `interface{}` would be an API improvement, but the reflection required to turn an `interface{}` into its underlying `[]interface{}` would be a performance hit within the driver here. Given the precedence of the `InsertMany` API, I think it's ok to keep this as-is.